src/popup/stores/AddTokenStore.ts

```diff
@@ -0,0 +1,109 @@
+import { observable, computed, action, reaction } from 'mobx';
+import { findIndex } from 'lodash';
+
+import AppStore from './AppStore';
+import { MESSAGE_TYPE } from '../../constants';
+import { isValidContractAddressLength } from '../../utils';
+
+const INIT_VALUES = {
+  contractAddress: '',
+  name: '',
+  symbol: '',
+  decimals: undefined,
+  getQRCTokenDetailsFailed: false,
+};
+
+export default class AddTokenStore {
+  @observable public contractAddress?: string;
+  @observable public name?: string;
+  @observable public symbol?: string;
+  @observable public decimals?: number;
+  @observable public getQRCTokenDetailsFailed?: boolean;
```

Comment:
use the INIT_VALUES. also, you are checking for `undefined` in the components for the values so technically you should change the init values to `undefined` (except for `getQRCTokenDetailsFailed`)

Reply:
? I I am checking for undefined? which check are you referring to?
I have this check in my AddToken component
<Typography className={classes.detailValue}>{value || ''}</Typography>

Reply:
your `INIT_VALUES` for `name, symbol, decimals` (empty string) are different than the initial values here (undefined).

should set the default values as the `INIT_VALUES` and make the `INIT_VALUES.name, symbol, decimals = undefined` to match what the defaults are.

in the comp, you are checking `{value || ''}` which means you assume the default values are undefined.